src/stack.rs

```diff
@@ -0,0 +1,22 @@
+pub struct Stack<T> {
```

Comment:
Not sure if there's much benefit of this thin wrapper? At best it just hides that `pop` may fail.

Reply:
not sure I understand how it's hiding the `pop` failure? If `pop` fails it produces an error, also covered by [test case](https://github.com/Pavel-Durov/pavel.yaiwr/blob/virtual-machine/tests/stack.rs#L27) 

I added it only to remove multiple `.expect` calls:

![diff](https://user-images.githubusercontent.com/6065236/223154755-6013df25-3af1-452b-9dac-eed1fe0425e1.png)

But I don't have a strong opinion either way :)

Reply:
That's what I mean: your `pop()` hides the case where there is nothing to pop.

Your tests are checking for over/underflow, not pop.

I don't object to the stack wrapper, but I don't think it offers much.

Reply:
I'm pretty sure the test I linked is checking `pop` error:
```
    #[test]
    #[should_panic(expected = "empty stack")]
    fn empty_pop() {
        let mut stack: Stack<usize> = Stack::new();
        stack.push(42);
        stack.pop();
        stack.pop(); // <------ here's the pop error
    }
```
https://github.com/Pavel-Durov/pavel.yaiwr/blob/virtual-machine/tests/stack.rs#L27
:no_mouth:

Reply:
removed stack wrapper :point_right: https://github.com/softdevteam/pavel.yaiwr/pull/10/commits/b3e45550c2aaf0ebf89a4bd7b1418a226f02e4d6

Reply:
OK, I now see why you wrapped the stack.

What we have is OK for now, but perhaps another time we should make a custom error type so we don't have to duplicate those strings all the time.